build_tools/benchmarks/common/benchmark_driver.py

```diff
@@ -68,20 +68,24 @@ def run(self) -> None:
 
     use_legacy_name = self.benchmark_suite.legacy_suite
 
-    target_architectures = []
-    cpu_target_arch = self.device_info.get_iree_cpu_arch_name(use_legacy_name)
-    if cpu_target_arch is None:
-      print("WARNING: Detected unsupported CPU architecture in "
-            f'"{self.device_info}", CPU benchmarking is disabled.')
-    else:
-      target_architectures.append(cpu_target_arch)
+    if self.config.use_compatible_filter:
+      compatible_architectures = []
+      cpu_target_arch = self.device_info.get_iree_cpu_arch_name(use_legacy_name)
+      if cpu_target_arch is None:
+        print("WARNING: Detected unsupported CPU architecture in "
+              f'"{self.device_info}", CPU benchmarking is disabled.')
+      else:
+        compatible_architectures.append(cpu_target_arch)
 
-    gpu_target_arch = self.device_info.get_iree_gpu_arch_name(use_legacy_name)
-    if gpu_target_arch is None:
-      print("WARNING: Detected unsupported GPU architecture in "
-            f'"{self.device_info}", GPU benchmarking is disabled.')
+      gpu_target_arch = self.device_info.get_iree_gpu_arch_name(use_legacy_name)
+      if gpu_target_arch is None:
+        print("WARNING: Detected unsupported GPU architecture in "
+              f'"{self.device_info}", GPU benchmarking is disabled.')
+      else:
+        compatible_architectures.append(gpu_target_arch)
     else:
-      target_architectures.append(gpu_target_arch)
+      # No compatible filter on the target architectures.
```

Comment:
It seems like we could give the user better feedback here if they *didn't* pass `--compatible-only` but we detect an incompatible architecture. Two options:

1. This is a hard failure. First check if the arch is compatible. If it's not, then check for the flag. Give an info message or a hard error.
2. This is a warning. Maybe we want people to be able to run benchmarks on something that we don't detect as compatible (e.g. unknown uarch, but they know it will/might work). Do the same checks as 1, but instead of a hard error, issue a warning like

```python
        print(f"WARNING: Detected incompatible CPU archiecture architecture"
              f" '{self.device_info}'. Pass --compatible-only to skip"
              f" benchmarks.")
```

Reply:
Yeah, I'd like to make sure the tool can still run on unknown hardware. Changed it to compare the benchmarks with the detected architectures and print warnings if they don't match.